src/sid/simulate.py

```diff
@@ -895,7 +947,7 @@ def _process_initial_states(
     # Sort index for deterministic shuffling and reset index because otherwise it will
     # be dropped while writing to parquet. Parquet stores an efficient range index
     # instead.
-    states = states.sort_index().reset_index()
+    states = states.sort_index().reset_index(drop=True)
```

Comment:
I would suggest to warn the user in case the index of the passed states are not a RangeIndex:

```suggestion
    if not isinstance(states.index, pd.RangeIndex):
        warnings.warn("The index of states passed to sid are reset and dropped by default.")
    states = states.sort_index().reset_index(drop=True)
```